extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -94,14 +96,27 @@ class InstrumentationService {
     /** @private {!Observable<!Event>} */
     this.clickObservable_ = new Observable();
 
-    /** @private {!Object<string, !Observable<!AnalyticsEvent>>} */
-    this.observers_ = {};
-
     /** @private {boolean} */
     this.scrollHandlerRegistered_ = false;
 
     /** @private {!Observable<Event>} */
     this.scrollObservable_ = new Observable();
+
+    /** @private {!Object<string, !Observable<!AnalyticsEvent>>} */
+    this.customEventObservers_ = {};
+
+    /**
+     * Some early events have to be buffered because there's no way to predict
+     * how fast all `amp-analytics` elements will be instrumented.
+     * @private {!Array<!AnalyticsEvent>|undefined}
+     */
+    this.customEventBuffer_ = [];
+
+    // Stop buffering of custom events after 10 seconds. Assumption is that all
+    // `amp-analytics` elements will have been instrumented by this time.
+    timer.delay(() => {
```

Comment:
There's unfortunately nothing to check for - there are no conditions that would tell us that a new client should be pending.